### PR TITLE
Fixed #31558 -- Added support for boolean attribute on properties in ModelAdmin.list_display.

### PR DESCRIPTION
**Pull request by RyanWalker277**
Fixes [#31556](https://code.djangoproject.com/ticket/31558)

Refs #17092

### CODE REVIEW

File: docs/ref/contrib/admin/index.txt [hunk 1]
```diff
@@ -816,6 +816,7 @@ subclass::
 
           my_property.short_description = "Full name of the person"
           my_property.admin_order_field = "last_name"
+          my_property.boolean = True
```
Comment (sarahboyce) [inline docs/ref/contrib/admin/index.txt:819]: If you run `make black` in the docs directory, that should fix the failing blacken docs GitHub action.
Should we put `my_property.boolean = False` here as this example of a boolean doesn't make sense? Maybe also:
```python
              @admin.display(
                  ordering="last_name",
                  description="Full name of the person",
                  boolean=False,
              )
              def full_name(self):
                  return self.first_name + " " + self.last_name
```
Might want to add a `.. versionadded:: 5.0` note (or 5.1 if we don't get it in in time) that specifically draws attention to the new boolean property and what it does

I think we maybe want a small release note within the `django.contrib.admin` section

File: tests/modeladmin/models.py [hunk 2]
```diff
@@ -54,3 +54,19 @@ def decade_published_in(self):
...
+    def __str__(self):
+        return self.name
+
+    def my_property(self):
+        return self.is_active
+
+    my_property.boolean = True
+    my_property.short_description = "Is Active?"
+    my_property.admin_order_field = "name"
+    my_property = property(my_property)
```
Comment (sarahboyce) [inline tests/modeladmin/models.py:72]: This model is not required and can be removed

**Review by sarahboyce — COMMENTED** [review]
Thank you for this patch @RyanWalker277! :tada: 
I have some minor comments :star:

File: django/contrib/admin/utils.py [hunk 3]
```diff
@@ -296,6 +296,12 @@ def lookup_field(name, obj, model_admin=None):
         elif hasattr(model_admin, name) and name != "__str__":
             attr = getattr(model_admin, name)
             value = attr(obj)
+        elif hasattr(model_admin, "model") and hasattr(model_admin.model, name):
+            attr = getattr(model_admin.model, name)
+            try:
+                value = getattr(obj, name)()
+            except TypeError:
+                value = getattr(obj, name)
```
Comment (felixxm) [inline django/contrib/admin/utils.py:304]: I think we can simplify this:
```diff
diff --git a/django/contrib/admin/utils.py b/django/contrib/admin/utils.py
index 70f63d3cfe..d6c54dd440 100644
--- a/django/contrib/admin/utils.py
+++ b/django/contrib/admin/utils.py
@@ -296,18 +296,14 @@ def lookup_field(name, obj, model_admin=None):
         elif hasattr(model_admin, name) and name != "__str__":
             attr = getattr(model_admin, name)
             value = attr(obj)
-        elif hasattr(model_admin, "model") and hasattr(model_admin.model, name):
-            attr = getattr(model_admin.model, name)
-            try:
-                value = getattr(obj, name)()
-            except TypeError:
-                value = getattr(obj, name)
         else:
             attr = getattr(obj, name)
             if callable(attr):
                 value = attr()
             else:
                 value = attr
+            if hasattr(model_admin, "model") and hasattr(model_admin.model, name):
+                attr = getattr(model_admin.model, name)
         f = None
     else:
         attr = None
```

File: docs/ref/contrib/admin/index.txt (hunk 1 above)
Comment (felixxm) [reply docs/ref/contrib/admin/index.txt:819]: > I think we maybe want a small release note within the `django.contrib.admin` section

Yes, for example:
```
* Properties in :attr:`.ModelAdmin.list_display` now support ``boolean``
  attribute.

**Comment by felixxm** [comment]
@RyanWalker277 Thanks :+1: Welcome aboard :boat: 

I pushed final edits.

**Pull request merged**